Lib/test/test_linecache.py

```diff
@@ -374,5 +376,40 @@ def test_checkcache_with_no_parameter(self):
         self.assertIn(self.unchanged_file, linecache.cache)
 
 
+class FreeThreadingTest(unittest.TestCase):
```

Comment:
This is a general thread-safety test. It's not really specific to the free threaded build.